int32 delta_x
int32 delta_y
float32 global_x
float32 global_y
int32 color
bool found